wakuv2/waku.go

```diff
@@ -218,17 +218,26 @@ func New(nodeKey string, fleet string, cfg *Config, logger *zap.Logger, appDB *s
 		onHistoricMessagesRequestFailed: onHistoricMessagesRequestFailed,
 		onPeerStats:                     onPeerStats,
 	}
+	enablePeerExchange := false
+	enableDiscv5 := false
+	if cfg.LightClient {
+		enablePeerExchange = true
```

Comment:
This enables server side of peerExchange which is not required in case of LightClient. 
But in relay-mode where discv5 is enabled, we may want to enable this. 

We need to make sure peerExchange client is running in lightMode, which i did not see any setting here. Rather the start of peerExchangeLoop happens.

Reply:
If we want to have different configs to run peerExchange client and server that is also fine. But right now it looks like same config is being used for both which doesn't make sense. 

PeerExchange server makes sense only when discv5 is running locally as it will have a cache of peers that it can respond to in case a PeerExchange request comes from the PeerExchangeClient.

Also looks like some of my changes to consider bootNodes as peerExchange peers have been over-ridden which must be causing the issue reported in mobile-CI where there are no nodes identified for lightpush/filter.

Reply:
The solution is to add the bootnodes as peers and let identify protocol run (by inviking `identifyAndConnect` method. Post this, the peerExchange loop will start functioning properly and should be able to discover peers.

Reply:
We can remove following lines  and invoke 
`go w.identifyAndConnect(w.ctx, true, discoveredNode.PeerInfo)`

https://github.com/status-im/status-go/pull/4665/files#diff-2dce231f3b66e8e19264149a6034a5832c56ce56805789296fd27f38eef8def1R595-R605

Reply:
cc @vitvly

Reply:
Already addressed as part of another PR https://github.com/status-im/status-go/pull/4826 , so we dont need these changes anymore.